### DISCUSSION TITLE
Oven modeling

### DISCUSSION THREAD

**Discussion by ErneD** [frePPLe/frepple#601] [Q&A] [answered]
Hi !

I am trying to model a oven for ceramics.

Each batch cooked takes 2 days of the oven and 1 day of an operator.
The operation takes allways 2 days BUT the volume capacity of the oven isn't unconstrained: for item A the capacity is 10 eaches and for item B capacity is 2.
I have managed to model this using a bucket resource (the oven) with a maximum calendar holding the number of units to be consumed per day. And a default resource (operator)
I  'm facing two issues that I can't solve:

1. The capacity of the oven, even if it is a 2 day operation, is consumed only for the first bucket, leaving the oven available after 1 day. **How could I configure the system so the resource is unavailable during the 2 days ?**

2. When the oven is used by only one operation (10 each of item A for instance) the operator resource is consumed by this operation, no problem there. The issue is when there are 2 operations in the oven (5 item A and 1 item B) because these trigger the request for 2 operators, giving an impossible situation. 
**How can I configure the system so the oven request only 1 operator regardless of the operations being produced in the oven ? And this for only 1 of the 2 days of cooking ?**

Can some please help me with this ?

Thank you !

Ernesto.

**Comment by jdetaeye** [comment, marked as answer]
Modeling ovens is tricky (not only in frepple, but in any planning tool). We currently have only an approximate modeling for it.

> 1. The capacity of the oven, even if it is a 2 day operation, is consumed only for the first bucket, leaving the oven available after 1 day. How could I configure the system so the resource is unavailable during the 2 days ?

That's only possible with a default resource.  Instead of playing with the resource capacity, you need to work with quantity of the operation-resource records.
Eg:   Resource "oven" has a size of "3" (which could be the square meter of the oven surface).  When an operation uses 0.25 units of the resource, that resource can do 12 of these jobs in parallel.

You'll need to use operations with "fixed-time" and also set the operation "size-maximum" field to break a big requirement into a sequence of "oven batches".  

>>When the oven is used by only one operation (10 each of item A for instance) the operator resource is consumed by this operation, no problem there. The issue is when there are 2 operations in the oven (5 item A and 1 item B) because these trigger the request for 2 operators, giving an impossible situation.
>>How can I configure the system so the oven request only 1 operator regardless of the operations being produced in the oven ? And this for only 1 of the 2 days of cooking ?

Operator usage for ovens is also a bit tricky. Your operator is probably only needed for loading and unloading of the oven, and not during the complete baking time. Right?
In most situations we don't model that in detail. The planner can roughly estimate the number of operators based on the loading of the oven resources.  

Hope this helps,

Johan

**Reply by ErneD** [reply]
Hi Johan,

Thank you for your quick answer, I really appreciate it, and please accept my apologies for the delay in writing you back.

If I understand correctly your proposition is to have 1 operation per item, consumig the space occupied in the oven  right ? So the number of paralell operations is proportional to the space occupied in the oven right ?

I have tested it and it works, the oven is occupied during the whole 2 days of the batch ! (The only downside is that I get a lot of manufacturing orders because there is 1 opeation per item).

I agree with what you say regarding the operators required for the load, they can be scheduled afterwards.

I have one extra constrain that is giving me headackes.
The oven batch operation lasts 2 days and it is divided as follows

- 1 day of cooking (which consumes electricity)
- 1 day of cooling (which doesn't consumes electricity)

There are three ovens, but they can't cook at the same time due to electricity constraints. One oven can cook while the other is cooling but only one can cook at a time.

Is there a way to model this in Frepple ?

Thanks again for your time and your help Johan !
